src/content/reference/react/useOptimistic.md

```diff
@@ -0,0 +1,212 @@
+---
+title: useOptimistic
+canary: true
+---
+
+<Canary>
+
+The `useOptimistic` Hook is currently only available in React's canary and experimental channels. Learn more about [React's release channels here](/community/versioning-policy#all-release-channels).
+
+</Canary>
+
+<Intro>
+
+`useOptimistic` is a React Hook that lets you optimistically update the UI.
+
+```js
+  const [optimistic, addOptimistic] = useOptimistic(initialState, updateFn);
+```
+
+</Intro>
+
+<InlineToc />
+
+---
+
+## Reference {/*reference*/}
+
+### `useOptimistic(initialState, updateFn)` {/*use*/}
+
+`useOptimistic` is a React Hook that lets you optimistically update the UI.
```

Comment:
A few more sentences to outline how it accomplishes this, something like:

It accepts some state as an argument and returns a copy of that state that can be different during the duration of an async action such as a network request. You provide a function that takes the current state and the input to the action, and returns the _optimistic_ state to be used while the action is pending.

This state is called the "optimistic" state because it is usually used to immediately present the user with the result of performing an action, even though the action actually takes time to complete.

Reply:
It is kinda redundant to say the same thing as the top, I think this place could be a better place to expand on what the hook does re Dave's point.

Reply:
Will update with the description